### PR TITLE
How to Build Templates for Django Applications with HTMX

### LINKED ISSUE #6936: How to Build templates for Django Applications with HTMX
### Writing sample(s): 
https://www.section.io/engineering-education/authors/muhammed-ali/

#### Proposed title of article
How to Build templates for Django Applications with HTMX

#### Proposed article introduction
In this article, I will build a simple Django application (API), create views of it then go into details of how the reader can implement HTMX in their Django frontend. This is important if you don't want to go through the stress of using a library like React or Vue.

ONLY viable topics to pick from:
- API
- Languages

#### Key takeaways 
- Learn about HTMX
- Learn how to connect HTMX with Django
- Learn how to fetch data directly from the Django API with HTMX

#### Article quality
it is a unique article.

### PR DESCRIPTION
**Pull request by khabdrick**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes
This closes #6936

### CODE REVIEW

**Comment by louisefindlay23** [comment]
Plagiarism check complete. Please remember to check your article for typos and formatting issues.

File: content/articles/django-htmx/index.md [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+### Introduction
+
+Did you know that it is possible to use AJAX without writing a single line of JavaScript code?
+
+Are you a Django developer that is not really familiar with JavaScript and will like to display components of your application asynchronously?
+
+If these questions have come to your mind, you are in the right place.
+
+In this article, I will build a simple Django application that creates, deletes content from the database, and displays the current content asynchronously without the page refreshing. This is important if you don't want to go through the stress of using a library like React or Vue.
+The project built inthis tutorial can be found on [GitHub](https://github.com/khabdrick/django-htmx-tutorial).
```
Comment (mohamedgh16) [inline content/articles/django-htmx/index.md:12]: Spelling error here.

File: content/articles/django-htmx/index.md [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+
+#### Prerequisites
+
+- Basic understanding of Django
+- Django installed(v3.2)
+- Python installed(v3.8)
```
Comment (mohamedgh16) [inline content/articles/django-htmx/index.md:18]: Remove all space lines between the header and the content.

File: content/articles/django-htmx/index.md [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+### Creating, Listing, and Deleting with HTMX and Django
+
+In this section, you will learn how to build the basic create, list and delete functionality with HTMX and Django to illustrate how HTMX in Django. We will be building a contact list application.
+
+Something to note, when working with HTMX if you must return something from the server-side it must be HTML fragments, not JSON.
+
+Let’s get right into it!
+
+First, let’s create an app for our contacts. You can do this by going to the root of your application and running the following command:
+
```
Comment (mohamedgh16) [inline content/articles/django-htmx/index.md:33]: Fix the punctuation errors in this section.

File: content/articles/django-htmx/index.md [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+urlpatterns = [
+    path('admin/', admin.site.urls),
+    path('create-contact/', create_contact, name='create-contact'),
+    path("contacts/", ContactList.as_view(), name='contact-list'),
+
+]
+```
+
+Let’s now create templates for the form and the contact list. To do this, go to the directory for `app` and create a new directory called *templates/* (the name of the file is mandatory). In the directory you just created, create a file called *base.html, contact.html, contact-list.html.* Now, paste the code below in the *base.html* file you just created. The code below contains all the **Content Delivery Network** (CDN) required for the whole project to display properly.
```
Comment (mohamedgh16) [inline content/articles/django-htmx/index.md:121]: Fix any punctuation errors here.

File: content/articles/django-htmx/index.md [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+    <hr/>
+    
+    <div id="contact-list">
+        {% include 'contact-list.html' %}
+    </div>
+</div>
+{% endblock content %}
+```
+
+Now let’s create the template to display the list of contacts. Paste the code below in the  `contact-list.html`
```
Comment (mohamedgh16) [inline content/articles/django-htmx/index.md:194]: Missing "."

**Review by ghost — COMMENTED** [review]
Great article @khabdrick , implement the changes so can move to tech review.

**Comment by khabdrick** [comment]
@mohamedgh16 I'm done with the corrections

File: content/articles/django-htmx/index.md [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+```html
+{% if contacts %}
+
+{% csrf_token %}
+<ul class="list-group col-4">
+{% for contact in contacts %}
+    <li class="list-group-item d-flex justify-content-between align-items-center">
+        {{ contact.name }}: {{ contact.phone_number }}
+        
+
```
Comment (mohamedgh16) [inline content/articles/django-htmx/index.md:201]: Please remove any extra space lines inside or outside of the code section.

Comment (mohamedgh16) [inline content/articles/django-htmx/index.md:201]: Please remove any extra space lines inside or outside of the code section.

**Comment by khabdrick** [comment]
@mohamedgh16 I have made the changes

**Comment by hectorkambow** [comment]
Noted: just going through the queue in order - we will get to this one asap.!

**Comment by hectorkambow** [comment]
Jumping on this one

**Pull request merged**